### PR TITLE
Add user page

### PR DESCRIPTION
**Pull request by kyudori**
# Description

LPVS Front-End User Page. (logout, modify user information)
History, Result, Dashboard pages will be uploaded.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Code cleanup/refactoring
- [ ] Documentation update
- [x] This change requires a documentation update
- [ ] CI system update
- [ ] Test Coverage update

# How Has This Been Tested?
For using social login with Naver, Kakao, and Google, fill the necessary information:

- Update the `application.properties` located in `src/main/resources`:
Enter client-id and client-secret. (Kakao doesn't require a client-secret.)

- Update the `.env` file located in `frontend`:
Enter your REST_API_KEY.

The values required to implement the login function can be obtained from the link below.
Naver: https://developers.naver.com/products/login/api/api.md
Kakao: https://developers.kakao.com/docs/latest/ko/kakaologin/common
Google: https://cloud.google.com/identity-platform/docs/web/google

- frontend
```
cd frontend
npm install
npm start
```

- backend
```
mvn clean install
cd target
java -jar lpvs-*.jar
```
1. Use Chrome to navigate to http://localhost:3000
2. On the Home Page, click Login in the upper right corner.
3. You can login to any of the three places you want, Naver, Kakao, and Google.
4. After logging in, click your name in the upper right corner of the Home Page to go to the User Page. 
(If the name is long, it is abbreviated.)
5. You can type GitHub ID or Organization by pressing the edit button on the User page.
user
6. When the input is complete, click the Admit button to save entered input to the DB.
(If the Admit function worked normally, output the information about the reflected content as an alert.)
admit
(The GitHub ID must be entered; otherwise, a warning alert will be output.)
duplicate

**Test Configuration**:
* Java: v11
* LPVS Release: v1.1.0

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Review by tiokim — APPROVED** [review]
LGTM

File: src/main/java/com/lpvs/controller/LPVSWebController.java [hunk 1]
```diff
@@ -38,14 +39,12 @@ public LPVSWebController(LPVSMemberRepository memberRepository, LPVSDetectedLice
         this.licenseRepository = licenseRepository;
         this.lpvsLoginCheckService = LPVSLoginCheckService;
     }
-
```
Comment (m-rudyk) [inline src/main/java/com/lpvs/controller/LPVSWebController.java:41]: Please keep empty line for separation

File: src/main/java/com/lpvs/controller/LPVSWebController.java [hunk 2]
```diff
@@ -38,14 +39,12 @@ public LPVSWebController(LPVSMemberRepository memberRepository, LPVSDetectedLice
...
     @GetMapping("user/info")
     @ResponseBody
     public LPVSMember personalInfoSettings(Authentication authentication) {
         lpvsLoginCheckService.loginVerification(authentication);
         return lpvsLoginCheckService.getMemberFromMemberMap(authentication);
     }
-
```
Comment (m-rudyk) [inline src/main/java/com/lpvs/controller/LPVSWebController.java:48]: here as well

File: src/main/java/com/lpvs/auth/SecurityConfig.java [hunk 3]
```diff
@@ -33,7 +33,7 @@ public SecurityFilterChain filterChain(HttpSecurity http) throws Exception {
                 .and()
                 .logout()
                 .logoutRequestMatcher(new AntPathRequestMatcher("/oauth/logout"))
-                .logoutSuccessUrl("/")
+                .logoutSuccessUrl("http://localhost:3000")
```
Comment (m-rudyk) [inline src/main/java/com/lpvs/auth/SecurityConfig.java:36]: Maybe we need to move this value to application.properties?

Comment (o-kopysov) [reply src/main/java/com/lpvs/auth/SecurityConfig.java:36]: Agree. If we run FE on the server, a hardcoded address could be a problem.

File: src/main/java/com/lpvs/controller/LPVSWebController.java [hunk 4]
```diff
@@ -60,4 +59,19 @@ public LPVSLoginMember loginMember(Authentication authentication) {
         }
     }
 
+    @CrossOrigin(origins = "http://localhost:3000")
```
Comment (m-rudyk) [inline src/main/java/com/lpvs/controller/LPVSWebController.java:62]: Please consider to set this in properties.

File: src/main/java/com/lpvs/exception/PageControllerAdvice.java [hunk 5]
```diff
@@ -26,4 +26,11 @@ public ResponseEntity<ErrorResponse> loginFailedHandle(LoginFailedException e) {
         return new ResponseEntity<>(errorResponse, HttpStatus.UNAUTHORIZED);
     }
 
+    @ExceptionHandler(IllegalArgumentException.class)
+    public ResponseEntity<ErrorResponse> handleSQLException(IllegalArgumentException e) {
+        log.info("duplicated key exception" + e.getMessage());
```
Comment (m-rudyk) [inline src/main/java/com/lpvs/exception/PageControllerAdvice.java:31]: How about changing log level `info` -> `error`?

**Review by m-rudyk — APPROVED** [review]
Approved. Please change few remarks.

File: src/main/java/com/lpvs/exception/PageControllerAdvice.java (hunk 5 above)
Comment (kyudori) [reply src/main/java/com/lpvs/exception/PageControllerAdvice.java]: Done.

File: src/main/java/com/lpvs/controller/LPVSWebController.java (hunk 4 above)
Comment (kyudori) [reply src/main/java/com/lpvs/controller/LPVSWebController.java:64]: Done.

File: src/main/java/com/lpvs/controller/LPVSWebController.java (hunk 2 above)
Comment (kyudori) [reply src/main/java/com/lpvs/controller/LPVSWebController.java]: Done.

File: src/main/java/com/lpvs/controller/LPVSWebController.java (hunk 1 above)
Comment (kyudori) [reply src/main/java/com/lpvs/controller/LPVSWebController.java]: Done.

File: src/main/java/com/lpvs/auth/SecurityConfig.java (hunk 3 above)
Comment (kyudori) [reply src/main/java/com/lpvs/auth/SecurityConfig.java]: Moved the value to application.properties.
I'd appreciate it if you could check if the name of the newly added variable is okay.

File: src/main/java/com/lpvs/controller/LPVSWebController.java [hunk 6]
```diff
@@ -60,4 +61,19 @@ public LPVSLoginMember loginMember(Authentication authentication) {
...
```
Comment (o-kopysov) [inline src/main/java/com/lpvs/controller/LPVSWebController.java:64]: @kyudori  Why don't you use `corsAllowedOrigin` here?

**Review by o-kopysov — APPROVED** [review]
LGTM

**Pull request merged**